data_model/src/events/mod.rs

```diff
@@ -134,23 +160,25 @@ pub trait Filter {
 
 #[cfg(feature = "transparent_api")]
 impl Filter for FilterBox {
-    type Event = Event;
+    type Event = EventBox;
```

Comment:
Why are you introducing the `Box` naming here? I though the main reason to use it was to make sure the enum name doesn't conflict with the corresponding trait name. With events, however, I do not think such a problem exists?